### PR TITLE
[pkgconf] Update to 2.1.0

### PR DESCRIPTION
**Pull request by dg0yt**
2.0 version were unusable due to https://github.com/pkgconf/pkgconf/issues/326,  https://github.com/pkgconf/pkgconf/issues/322. The fixes were developed in the context of, and tested in, this PR.

### CODE REVIEW

**Comment by dg0yt** [comment]
Maybe we also need to look at https://github.com/pkgconf/pkgconf/issues/322.

File: ports/jkqtplotter/portfile.cmake [hunk 1]
```diff
@@ -26,7 +26,7 @@ vcpkg_cmake_configure(
         -DJKQtPlotter_BUILD_SHARED_LIBS=${JKQtPlotter_BUILD_SHARED_LIBS}
         -DJKQtPlotter_BUILD_STATIC_LIBS=${JKQtPlotter_BUILD_STATIC_LIBS}
         ${FEATURE_OPTIONS}
-        -DQT_VERSION_MAJOR=6
+        -DCMAKE_IGNORE_PATH=${CURRENT_INSTALLED_DIR}/share/cmake/Qt5
```
Comment (dg0yt) [inline ports/jkqtplotter/portfile.cmake:29]: @Neumann-A The last build failed with 
~~~
CMake Error at C:/a/1/s/scripts/buildsystems/vcpkg.cmake:859 (_find_package):
  Could not find a package configuration file provided by "Qt5Svg" with any
  of the following names:

    Qt5SvgConfig.cmake
    qt5svg-config.cmake

  Add the installation prefix of "Qt5Svg" to CMAKE_PREFIX_PATH or set
  "Qt5Svg_DIR" to a directory containing one of the above files.  If "Qt5Svg"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  D:/installed/x64-windows-static/share/cmake/Qt5/Qt5Config.cmake:28 (find_package)
  C:/a/1/s/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
  cmake/jkqtplotter_common_qtsettings.cmake:8 (find_package)
  CMakeLists.txt:40 (include)
~~~

~~~
//The directory containing a CMake configuration file for QT.
QT_DIR:PATH=D:/installed/x64-windows-static/share/cmake/Qt5
~~~

**Comment by dg0yt** [comment]
Included in https://github.com/microsoft/vcpkg/pull/35319 (merged) and #35331 (proposed).

**Comment by dg0yt** [comment]
Ping for review.

(Next upstream fix pending, but not yet approved: https://github.com/pkgconf/pkgconf/pull/336)

**Review by LilyWangLL — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**